### PR TITLE
Dev general representation for neural network

### PR DESCRIPTION
**Pull request by Crysple**

### CODE REVIEW

File: examples/trials/ir_enas/config.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    controller_train_steps: 1
+
+trial:
+  command: sh ./macro_cifar10.sh
+  codeDir: ./
+  gpuNum: 1
+
+# machineList:
+#   - ip: 10.150.144.121
+#     username: v-zejlin
```
Comment (xuehui1991) [inline examples/trials/ir_enas/config.yml]: remove this?

File: examples/trials/ir_enas/config.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+authorName: Unknown
+experimentName: enas_macro
```
Comment (xuehui1991) [inline examples/trials/ir_enas/config.yml]: shall we put this example to a personal repo?

File: examples/trials/ir_enas/nni_controller_cifar10.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+os.environ['CUDA_VISIBLE_DEVICES'] = ""
```
Comment (xuehui1991) [inline examples/trials/ir_enas/nni_controller_cifar10.py]: ?

File: tools/nni_annotation/testcase/architecture_usercode/simple.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+def add_three(x):
+    return x+3
+
+def add_four(x):
+    return x+4
+
+def add_five(x):
+    return x+5
+
+
```
Comment (xuehui1991) [inline tools/nni_annotation/testcase/architecture_usercode/simple.py]: remove this blank line?

File: examples/trials/ir_enas/config.yml (hunk 2 above)
Comment (leelaylay) [reply examples/trials/ir_enas/config.yml]: > shall we put this example to a personal repo?

I suggest so. Shall we create a PR to `countif/enas_nni` which contains the example?

**Pull request closed without merging**